### ISSUE TITLE
Modify rule S6303: False positive for CfnDBInstance and CfnDBCluster with some properties (AWS CDK)

### ISSUE THREAD

**Issue by gaetan-ferry-sonarsource** [SonarSource/SonarJS#3637]
# What

Implementation ticket https://github.com/SonarSource/SonarJS/issues/3440 specify the detection behavior for RDS related construct that should enable at-rest encryption.

Current rule implementation does not take into account the some important parameters of the construct properties:
* For `CfnDBCluster`:
  * `sourceDbClusterIdentifier`
  * `snapshotIdentifier`
* For `CfnDBInstance`
  * `sourceDbInstanceIdentifier`
  * `dbSnapshotIdentifier`

When any of those parameters is set, the encryption status is inherited from a source DB instance, cluster or snapshot. Thus, no issue should be raised.

# Example

```typescript
const cdbi1 = new rds.CfnDBInstance(this, 'SourceDBInstanceDelegatedCompliantCfnDatabaseInstance', {
    engine: 'mariadb',
    sourceDbInstanceIdentifier: 'XXX', // Compliant
});
```

```typescript
const cdbi1 = new rds.CfnDBInstance(this, 'SourceDBInstanceDelegatedCompliantCfnDatabaseInstance', {
    engine: 'mariadb',
    snapshotIdentifier: 'XXX', // Compliant
});
```

```typescript
const cdbc1 = new rds.CfnDBCluster(
    this,
    'ClusterDelegatedCompliantCfnDBCluster', { 
        engine: 'aurora',
        sourceDbClusterIdentifier: 'XXX', // Compliant
});
```

**Comment by Wohops** [comment]
Migrated to Jira. New ticket number: JS-103
